src/toil_lib/programs.py

```diff
@@ -121,6 +195,37 @@ def _fix_permissions(base_docker_call, tool, work_dir):
     :param str work_dir: Path of work directory to recursively chown
     """
     base_docker_call.append('--entrypoint=chown')
+    # We don't need the cleanup container to persist.
+    base_docker_call.append('--rm')
     stat = os.stat(work_dir)
     command = base_docker_call + [tool] + ['-R', '{}:{}'.format(stat.st_uid, stat.st_gid), '/data']
     subprocess.check_call(command)
+
+
+def _get_container_name(job):
+    return '--'.join([job.fileStore.jobStore.config.workflowID,
+                      job.fileStore.jobID,
+                      base64.b64encode(os.urandom(9), '-_')])
+
+
+def _container_is_running(container_name):
+    """
+    Checks whether the container is running or not.
+
+    :param container_name: Name of the container being checked.
+    :returns: True if running, False if not running, None if the container doesn't exist.
+    :rtype: bool
+    """
+    try:
+        output = subprocess.check_output(['docker', 'inspect', '--format', '{{.State.Running}}',
+                                          container_name]).strip()
+        if output == 'true':
+            return True
+        elif output == 'false':
+            return False
+        else:
+            raise RuntimeError("Got unexpected value for State.Running (%s)", output)
```

Comment:
This doesn't work by the way. Please just use my version. If my version is incorrect, let me know.